apis/management.cattle.io/v3/alerting_types.go

```diff
@@ -0,0 +1,159 @@
+package v3
+
+import (
+	"github.com/rancher/norman/types"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+type ClusterAlert struct {
+	types.Namespaced
+
+	metav1.TypeMeta `json:",inline"`
+	// Standard object’s metadata. More info:
+	// https://github.com/kubernetes/community/blob/master/contributors/devel/api-conventions.md#metadata
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec ClusterAlertSpec `json:"spec"`
+	// Most recent observed status of the cluster. More info:
+	// https://github.com/kubernetes/community/blob/master/contributors/devel/api-conventions.md#spec-and-status
+	Status AlertStatus `json:"status"`
+}
+
+type ProjectAlert struct {
+	types.Namespaced
+
+	metav1.TypeMeta `json:",inline"`
+	// Standard object’s metadata. More info:
+	// https://github.com/kubernetes/community/blob/master/contributors/devel/api-conventions.md#metadata
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec ProjectAlertSpec `json:"spec"`
+	// Most recent observed status of the cluster. More info:
+	// https://github.com/kubernetes/community/blob/master/contributors/devel/api-conventions.md#spec-and-status
+	Status AlertStatus `json:"status"`
+}
+
+type AlertCommonSpec struct {
+	DisplayName           string      `json:"displayName,omitempty" norman:"required"`
+	Description           string      `json:"description,omitempty"`
+	Severity              string      `json:"severity,omitempty" norman:"required,options=info|critical|warning,default=critical"`
+	Recipients            []Recipient `json:"recipients,omitempty" norman:"required"`
+	InitialWaitSeconds    int         `json:"initialWaitSeconds,omitempty" norman:"required,default=180,min=0"`
+	RepeatIntervalSeconds int         `json:"repeatIntervalSeconds,omitempty"  norman:"required,default=3600,min=0"`
+}
+
+type ClusterAlertSpec struct {
+	AlertCommonSpec
+
+	ClusterName         string              `json:"clusterName" norman:"type=reference[cluster]"`
+	TargetNode          TargetNode          `json:"targetNode,omitempty"`
+	TargetSystemService TargetSystemService `json:"targetSystemService,omitempty"`
+	TargetEvent         TargetEvent         `json:"targetEvent,omitempty"`
+}
+
+type ProjectAlertSpec struct {
+	AlertCommonSpec
+
+	ProjectName    string         `json:"projectName" norman:"type=reference[project]"`
+	TargetWorkload TargetWorkload `json:"targetWorkload,omitempty"`
+	TargetPod      TargetPod      `json:"targetPod,omitempty"`
+}
+
+type Recipient struct {
+	Recipient    string `json:"recipient,omitempty"`
+	NotifierID   string `json:"notifierId,omitempty" norman:"required,type=reference[notifier]"`
+	NotifierType string `json:"notifierType,omitempty" norman:"required,options=slack|email|pagerduty|webhook"`
+}
+
+type TargetNode struct {
+	ID           string            `json:"id,omitempty"`
```

Comment:
@vincent99 also not sure if this is user facing, but if it is, we can use `NodeName` and put norman type as a reference, so it will get converted to nodeId on the API layer. Here is an example: https://github.com/rancher/types/blob/master/apis/management.cattle.io/v3/authz_types.go#L98


The same for other fields referencing actual k8s resources